Controller/EmbedController.php

```diff
@@ -19,7 +20,11 @@ public function __construct(ClientRequestManager $manager)
 
     public function renderHierarchyAction(string $template, string $parent, string $field, int $depth = null, array $sourceFields = [], array $args = []): Response
     {
-        $hierarchy = $this->clientRequest->getHierarchy($parent, $field, $depth, $sourceFields);
+        $activeChild = '';
+        if(isset($args['activeChild']) && $args['activeChild'] instanceof EMSLink) {
+            $activeChild = $args['activeChild']->getOuuid();
+        }
+        $hierarchy = $this->clientRequest->getHierarchy($parent, $field, $depth, $sourceFields, $activeChild);
```

Comment:
renderHierarchy is called from a twig function. The twig function will send an Ouuid. (we cannot create an EMSLink object in twig).

so $args['activeChild'] will be a simple string, that is in fact the Ouuid. (So your code will be much simpler.):

$hierarchy = $this->clientRequest->getHierarchy($parent, $field, $depth, $sourceFields, $args['activeChild'] ?? null);